### PR TITLE
Fix race condition in test

### PR DESCRIPTION
**Pull request by kegsay**
Implements half of #243 in a different way which involves less error prone locking.

### CODE REVIEW

File: sync2/device_data_ticker_test.go [hunk 1]
```diff
@@ -10,12 +10,31 @@ import (
...
+	s.slice = append(s.slice, item)
+}
+
+func (s *syncSlice[T]) clone() []T {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	result := make([]T, len(s.slice))
+	copy(result, s.slice)
+	return result
+}
```
Comment (DMRobertson) [inline sync2/device_data_ticker_test.go:30]: is there no sync.Slice the same way there is sync.Map?! odd

File: sync2/device_data_ticker_test.go [hunk 2]
```diff
@@ -31,29 +50,31 @@ func TestDeviceTickerBasic(t *testing.T) {
 		DeviceID: "b",
 	})
 	time.Sleep(duration * 2)
-	if len(payloads) != 1 {
-		t.Fatalf("expected 1 callback, got %d", len(payloads))
+	result := payloads.clone()
```
Comment (DMRobertson) [inline sync2/device_data_ticker_test.go:53]: Why do we need to clone here, sorry?

**Review by DMRobertson — APPROVED** [review]

File: sync2/device_data_ticker_test.go (hunk 2 above)
Comment (kegsay) [reply sync2/device_data_ticker_test.go:53]: We'd need to lock every time we access the underlying slice, which could then lead to inconsistent results (e.g len=1 when len() is called, but then increases to len=2 later on. It's easier to snapshot the slice.

File: sync2/poller_test.go [hunk 3]
```diff
@@ -18,6 +18,31 @@ import (
...
+var (
+	timeSinceMu    sync.Mutex
+	timeSinceValue = time.Duration(0) // 0 means use the real impl
+)
+
+func setTimeSinceValue(val time.Duration) {
+	timeSinceMu.Lock()
+	timeSinceValue = time.Minute * 2
+	timeSinceMu.Unlock()
+}
```
Comment (DMRobertson) [inline sync2/poller_test.go:34]: val seems unused here?

**Pull request merged**